src/ff/ec_prime_field.rs

```diff
@@ -0,0 +1,255 @@
+    fn from(s: Fp25519) -> Self {
+        s.0
+    }
+}
+
+impl Serializable for Fp25519 {
+    type Size = <<Fp25519 as SharedValue>::Storage as Block>::Size;
+
+    fn serialize(&self, buf: &mut GenericArray<u8, Self::Size>) {
+        *buf.as_mut() = self.0.to_bytes();
+    }
+
+    fn deserialize(buf: &GenericArray<u8, Self::Size>) -> Self {
+        Fp25519(Scalar::from_bytes_mod_order((*buf).into()))
+    }
+}
+
+impl rand::distributions::Distribution<Fp25519> for rand::distributions::Standard {
+    fn sample<R: crate::rand::Rng + ?Sized>(&self, rng: &mut R) -> Fp25519 {
+        let mut scalar_bytes = [0u8; 32];
+        rng.fill_bytes(&mut scalar_bytes);
+        Fp25519(Scalar::from_bytes_mod_order(scalar_bytes))
+    }
+}
+
+impl std::ops::Add for Fp25519 {
+    type Output = Self;
+
+    fn add(self, rhs: Self) -> Self::Output {
+        Self(self.0 + rhs.0)
+    }
+}
+
+impl std::ops::AddAssign for Fp25519 {
+    #[allow(clippy::assign_op_pattern)]
+    fn add_assign(&mut self, rhs: Self) {
+        *self = *self + rhs;
+    }
+}
+
+impl std::ops::Neg for Fp25519 {
+    type Output = Self;
+
+    fn neg(self) -> Self::Output {
+        Self(self.0.neg())
+    }
+}
+
+impl std::ops::Sub for Fp25519 {
+    type Output = Self;
+
+    fn sub(self, rhs: Self) -> Self::Output {
+        Self(self.0 - rhs.0)
+    }
+}
+
+impl std::ops::SubAssign for Fp25519 {
+    #[allow(clippy::assign_op_pattern)]
+    fn sub_assign(&mut self, rhs: Self) {
+        *self = *self - rhs;
+    }
+}
+
+impl std::ops::Mul for Fp25519 {
+    type Output = Self;
+
+    fn mul(self, rhs: Self) -> Self::Output {
+        Self(self.0 * rhs.0)
+    }
+}
+
+impl std::ops::MulAssign for Fp25519 {
+    #[allow(clippy::assign_op_pattern)]
+    fn mul_assign(&mut self, rhs: Self) {
+        *self = *self * rhs;
+    }
+}
+
+impl From<Scalar> for Fp25519 {
+    fn from(s: Scalar) -> Self {
+        Fp25519(s)
+    }
+}
+
+#[cfg(test)]
+macro_rules! sc_hash_impl {
+    ( $u_type:ty) => {
+        impl From<Fp25519> for $u_type {
+            fn from(s: Fp25519) -> Self {
+                let hk = Hkdf::<Sha256>::new(None, s.0.as_bytes());
+                let mut okm = <$u_type>::MIN.to_le_bytes();
+                //error invalid length from expand only happens when okm is very large
+                hk.expand(&[], &mut okm).unwrap();
+                <$u_type>::from_le_bytes(okm)
+            }
+        }
+
+        impl From<$u_type> for Fp25519 {
+            fn from(s: $u_type) -> Self {
+                let hk = Hkdf::<Sha256>::new(None, &s.to_le_bytes());
+                let mut okm = [0u8; 32];
+                //error invalid length from expand only happens when okm is very large
+                hk.expand(&[], &mut okm).unwrap();
+                Fp25519::deserialize(&okm.into())
+            }
+        }
+    };
+}
+
+#[cfg(test)]
+sc_hash_impl!(u64);
+
+#[cfg(test)]
+sc_hash_impl!(u32);
+
+/// Daniel had to implement this since PRSS wants it, prefer not to
```

Comment:
can you open a quick issue to our repo and reference it here so we don't forget about it?